src/Weapon.ts

```diff
@@ -0,0 +1,50 @@
+import { Item } from "./Item";
+
+export class Weapon extends Item {
+  protected baseDamage: number;
+  protected damageModifier: number = 0;
+  protected baseDurability: number;
+  protected durabilityModifier: number = 0;
+  protected effectiveDurability: number;
+  protected isBroken: boolean = false;
+
+  static MODIFIER_CHANGE_RATE: number = 0.05;
+
+  constructor(name: string, baseDamage: number, baseDurability: number, value: number, weight: number ) {
+    super(name, value, weight);
+    this.baseDamage = baseDamage;
+    this.baseDurability = baseDurability;
+    this.effectiveDurability = this.getEffectiveDurability();
+  }
+
+  getEffectiveDamage() {
+    return this.baseDamage + this.damageModifier;
+  }
```

Comment:
add `abstract polish(): void;`

Reply:
according to tests Weapon can't be abstract class